src/Umbraco.Web.BackOffice/Controllers/ContentTypeController.cs

```diff
@@ -590,32 +590,41 @@ public ActionResult<ContentTypeImportModel> Upload(List<IFormFile> file)
 
                 var root = _hostingEnvironment.MapPathContentRoot(Constants.SystemDirectories.TempFileUploads);
                 var tempPath = Path.Combine(root,fileName);
-
-                using (var stream = System.IO.File.Create(tempPath))
-                {
-                    formFile.CopyToAsync(stream).GetAwaiter().GetResult();
-                }
-
-                if (ext.InvariantEquals("udt"))
+                if (Path.GetFullPath(tempPath).StartsWith(Path.GetFullPath(root)))
                 {
-                    model.TempFileName = Path.Combine(root, fileName);
+                    using (var stream = System.IO.File.Create(tempPath))
+                    {
+                        formFile.CopyToAsync(stream).GetAwaiter().GetResult();
+                    }
 
-                    var xd = new XmlDocument
+                    if (ext.InvariantEquals("udt"))
                     {
-                        XmlResolver = null
-                    };
-                    xd.Load(model.TempFileName);
+                        model.TempFileName = Path.Combine(root, fileName);
 
-                    model.Alias = xd.DocumentElement?.SelectSingleNode("//DocumentType/Info/Alias")?.FirstChild.Value;
-                    model.Name = xd.DocumentElement?.SelectSingleNode("//DocumentType/Info/Name")?.FirstChild.Value;
-                }
-                else
+                        var xd = new XmlDocument
+                        {
+                            XmlResolver = null
+                        };
+                        xd.Load(model.TempFileName);
+
+                        model.Alias = xd.DocumentElement?.SelectSingleNode("//DocumentType/Info/Alias")?.FirstChild.Value;
+                        model.Name = xd.DocumentElement?.SelectSingleNode("//DocumentType/Info/Name")?.FirstChild.Value;
+                    }
+                    else
+                    {
+                        model.Notifications.Add(new BackOfficeNotification(
+                            _localizedTextService.Localize("speechBubbles", "operationFailedHeader"),
+                            _localizedTextService.Localize("media", "disallowedFileType"),
+                            NotificationStyle.Warning));
+                    }
+                }else
```

Comment:
Just a nitpick, line break needed before the `else`?

Reply:
You have an eye for detail :)
As mentioned on the comment above, this PR was meant to only include the SECURITY.md change, since the other changes were part of #11537 and have been merged already.